### PR TITLE
feat(ui) : test portfolio roadmap

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Follow up PR to #53963 where the icons are analyzed. For some reason the certified user didn't register as complete for the english curriculum and I'm a little confused.

### CODE REVIEW

**Review by bader1379 — COMMENTED** [review]
Bader

**Review by bader1379 — COMMENTED** [review]
bader@bader

**Comment by moT01** [comment]
Yea, the certified user probably doesn't have the English challenges completed. Also, there's no certification for the English curriculum yet.

**Comment by a2937** [comment]
> Yea, the certified user probably doesn't have the English challenges completed. Also, there's no certification for the English curriculum yet.

Okay I'll go ahead and alter my tests to not include the English curriculum for the certified user.

**Review by naomi-lgbt — APPROVED** [review]

File: e2e/learn.spec.ts [hunk 1]
```diff
@@ -81,4 +81,45 @@ test.describe('Learn (authenticated user)', () => {
...
+
+  test('the page shows completed icons for a certified user', async function ({
+    page
+  }) {
+    await page.goto('/learn');
+    const curriculumBtns = page.getByTestId('curriculum-map-button');
+    await expect(curriculumBtns).toHaveCount(superBlocks.length);
+    for (let i = 0; i < superBlocks.length; i++) {
+      //Exclude English curriculum from tests and ones that lack circles; todo: eventually remove
+      console.log(i + ' \n' + superBlocks[i]);
```
Comment (ahmaxed) [inline e2e/learn.spec.ts:93]:
```suggestion
```

**Comment by ahmaxed** [comment]
This does not have to be a e2e test because it is testing a simple non-interactive functionality. But this should do that job. I left some comments for your review.

File: e2e/learn.spec.ts [hunk 2]
```diff
@@ -81,4 +81,45 @@ test.describe('Learn (authenticated user)', () => {
...
+      ) {
+        const btn = curriculumBtns.nth(i);
+        const icon = btn.locator('.progress-icon > svg').first();
+        await expect(icon).toHaveClass('completeIcon');
+      }
+    }
+  });
+});
+
+test.describe('Learn (unauthenticated user)', () => {
```
[the remark this answers is not in the document]
Comment (a2937) [reply e2e/learn.spec.ts:106]: I'm not sure what you mean here. Can you explain please?

**Comment by a2937** [comment]
Okay now I've got it.

**Comment by a2937** [comment]
The test is failing because the signed in user for whatever reason doesn't have those certs and I'm not sure how to fix it so the test user has the right data.

**Comment by Sembauke** [comment]
I will take a look @a2937!

**Comment by Sembauke** [comment]
Hey @a2937,

Unfortunately I have to close this because our certified user is not up to date. This means that the icons will never show up because most certifications are not completely finished.

Feel free to reopen this when the certified user is back up to date.

Sorry of the inconvenience

**Pull request closed without merging**